File: exercice2/scala/src/test/scala/fr/devoxx/devops/logs/sql/SparkSQL1Test.scala
package fr.devoxx.devops.logs.sql

import fr.devoxx.devops.logs.{SharedSparkSQLContext, SharedSparkContext}
import fr.devoxx.devops.logs.spark.Spark3
import org.scalatest.{FunSuite, Matchers}

class SparkSQL1Test extends FunSuite with SharedSparkSQLContext with Matchers {

  test("Les liens cassés") {
    val result = SparkSQL1(rdd, sqlContext).process
    result shouldBe 25

  }
}

File: exercice2/java/src/test/java/fr/devoxx/devops/logs/sql/SparkSQL1Test.java
package fr.devoxx.devops.logs.sql;

import fr.devoxx.devops.logs.SparkTest;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class SparkSQL1Test extends SparkTest {
    @Test
    public void test_spark() {
        SparkSQL1 test = new SparkSQL1();
        long result = test.process(rdd(), sqlContext());
        assertThat(result, is(25L));
    }
}
